public/tabs/chat/chat_page_content.tsx

```diff
@@ -174,50 +221,42 @@ export const ChatPageContent: React.FC<ChatPageContentProps> = React.memo((props
               onRegenerate={chatActions.regenerate}
               interaction={interaction}
             >
-              <MessageContent
-                message={message}
-                loading={
-                  isLatestOutput &&
-                  chatState.llmResponseType === LLMResponseType.STREAMING &&
-                  chatState.llmResponding
-                }
-              />
+              <MessageContent message={message} loading={isStreaming} />
             </MessageBubble>
-            {showSuggestions && <Suggestions message={message} inputDisabled={loading} />}
             <EuiSpacer />
-          </React.Fragment>
+            {showSuggestions && !isStreaming && (
+              <Suggestions message={message} inputDisabled={loading} />
+            )}
+          </div>
         );
       })}
-      {loading && chatState.llmResponseType === LLMResponseType.TEXT ? (
-        <>
-          <EuiSpacer />
-          <MessageBubble loading showActionBar={false} />
-        </>
-      ) : null}
-
-      {configSchema.chat.regenerateMessage &&
-        chatState.llmResponding &&
-        chatContext.conversationId && (
-          <div style={{ marginLeft: '8px', marginTop: 10 }}>
-            <EuiFlexGroup alignItems="flexStart" direction="column" gutterSize="s">
-              <EuiFlexItem>
-                <SuggestionBubble
-                  content="Stop generating response"
-                  color="default"
-                  iconType="crossInACircleFilled"
-                  onClick={() => chatActions.abortAction(chatContext.conversationId)}
-                />
-              </EuiFlexItem>
-            </EuiFlexGroup>
-          </div>
-        )}
+      <LoadingPlaceholder loading={isWaitingForRepsonse} height={messageSpacerHeight}>
+        {configSchema.chat.regenerateMessage &&
+          chatState.llmResponding &&
+          chatContext.conversationId && (
+            <div style={{ marginLeft: '8px', marginTop: 10 }}>
+              <EuiFlexGroup alignItems="flexStart" direction="column" gutterSize="s">
+                <EuiFlexItem>
+                  <SuggestionBubble
+                    content="Stop generating response"
+                    color="default"
+                    iconType="crossInACircleFilled"
+                    onClick={() => chatActions.abortAction(chatContext.conversationId)}
+                  />
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            </div>
+          )}
+      </LoadingPlaceholder>
       {chatState.llmError && (
-        <EuiEmptyPrompt
-          iconType="alert"
-          iconColor="danger"
-          title={<h2>Error from response</h2>}
-          body={chatState.llmError.message}
-        />
+        <div style={{ minHeight: messageSpacerHeight }}>
```

Comment:
Add minHeight for error case is a edge case, can we add a unit test to cover it?

Reply:
Done, add a test case to cover this line